### ISSUE TITLE
Responsive mode for small screens prevents access to sidebar content

### ISSUE THREAD

**Issue by demiankatz** [UniversalViewer/universalviewer#938]
**UV version:**
```
 universalviewer@v4.0.17
```

**I'm submitting a:**
- [x] bug report => please fork one of these [codesandbox examples](https://github.com/UniversalViewer/universalviewer/wiki/UV-Examples) with a repro of your issue and include a link to it below
- [ ] feature request => please use the [user stories repo](https://github.com/UniversalViewer/user-stories)
- [ ] support request => Please do not submit support requests here, use [stackoverflow](http://stackoverflow.com/questions/tagged/iiif)

**Current behavior:**
When the UV is displayed on a small screen (e.g. a mobile device or simply a very small browser window), the responsive behavior is hiding side panels without offering any way to access them. This is particularly problematic for things like the AV component when a multi-part item is involved (e.g. manifest https://api.bl.uk/metadata/iiif/ark:/81055/vdc_100133103553.0x000002/manifest.json ), because the left side panel with navigation between parts goes away, and there is no way to view anything except the first part.

A related issue is that the right side panel containing metadata also goes away. In the OpenSeadragon component, an info button appears at the bottom of the page which can be used to access metadata, but this info button is absent from other extensions (at very least, this was confirmed to be a problem in the AV component, but I suspect it may carry across to others as well).

**Expected behavior:**

All functionality should be accessible in mobile as well as via desktop mode.

**Steps to reproduce:**

Open manifest https://api.bl.uk/metadata/iiif/ark:/81055/vdc_100133103553.0x000002/manifest.json and shrink your browser window. You cannot change parts or view metadata.

**Comment by demiankatz** [comment]
I can confirm that this same problem impacts the PDF extension. See https://www.universalviewer.dev/#?xywh=-2688%2C-197%2C7943%2C3936&iiifManifestId=https%3A%2F%2Fdigital.library.villanova.edu%2FItem%2Fvudl%3A294631%2FManifest

**Comment by LlGC-szw** [comment]
+1 from NLW. We've had several complaints from users abou this.

**Comment by 404mike** [comment]
@demiankatz I've made the sidebars and header visible on mobile. No change made to the footer, should we make this visible too?

As this is a simple CSS change the is no change to the default behaviour of the onLoad of the UV which means that the left sidebar is expanded when the UV is loaded. Is this the behaviour that we want? should mobile view have all side panels collapsed onLoad? a separate issue/discussion possibly?

**Comment by demiankatz** [comment]
Thanks for getting the ball rolling on this, @404mike! This definitely seems like an improvement to me, but I agree that having the sidebars expanded by default on mobile is probably not desirable since it leaves very little real estate for the actual content. If that's easy to adjust, it might be worth sorting out at the same time as the rest... but if it's complicated, I'd be happy to accept some incremental progress and address it as a follow-up (though I defer to @edsilv's opinion, if he has one).

Regarding the footer, my institution doesn't currently leverage the search function, so the footer is not important to me personally... but others might feel differently. I'd welcome some more opinions on that piece!

**Comment by 404mike** [comment]
@demiankatz added a condition to not display the left sidebar if the viewport is below 767px which is the CSS breakpoint currently.

**Comment by mialondon** [comment]
Hi all! I've been discussing this with Scott Jenson (introduced in https://blogs.bl.uk/digital-scholarship/2024/11/collaborating-to-improve-usability-the-universal-viewer-project.html) and we were wondering what the most common mobile use cases / user stories / jobs to be done are.

Could folk who've heard from users struggling with the lack of sidebars share anything they know about why those users were on mobile devices, and/or what they were trying to do at the time?

**Comment by demiankatz** [comment]
At Villanova, we have two use cases that are impacted by the lack of sidebars problem:

1.) We have some digitized audio tapes, where each side of the tape is a separate track. In mobile mode, without the sidebars, users can only listen to side A -- there is no way to "turn over the tape."

2.) We have some collections of related PDFs presented as single manifests with multiple sections. Again, on mobile, users can only view the first part of the document, so quick and convenient mobile reading is not possible.

**Comment by mialondon** [comment]
Thanks @demiankatz! Do you know why those users are looking at items on mobile, rather than a device with a larger screen?

**Comment by demiankatz** [comment]
@mialondon, I can only speculate, but it's often more comfortable/convenient to listen to audio on a mobile device rather than a desktop. I think reading PDFs in mobile is less likely to be a preferred experience, though if you're doing a quick lookup, it can also just be a matter of convenience.

**Comment by demiankatz** [comment]
I'm starting to collect some technical details for future reference during implementation.

1.) The AV manifest in the original comment is no longer online. Here's a multi-part MediaElement extension example that might be useful: https://digital.library.villanova.edu/Item/vudl:283833/Manifest / https://www.universalviewer.dev/#?xywh=-1678%2C361%2C5471%2C3936&iiifManifestId=https%3A%2F%2Fdigital.library.villanova.edu%2FItem%2Fvudl%3A283833%2FManifest

2.) The mobile footer is completely missing from the MediaElement and PDF extensions -- so even the "more info" panel is inaccessible in those contexts.

3.) The existing mobile footer info button is implemented in [src/content-handlers/iiif/modules/uv-shared-module/FooterPanel.ts](https://github.com/UniversalViewer/universalviewer/blob/06504c1bbc5e65843d2ee01f1d6cddd1d96a14a7/src/content-handlers/iiif/modules/uv-shared-module/FooterPanel.ts). Follow the `$moreInfoButton` variable to find relevant logic. Note that there's an `updateMoreInfoButton()` function that's entirely commented out; presumably this logic has been reimplemented with CSS instead of JS.

**Comment by demiankatz** [comment]
I think there are a few obvious tasks we need to complete:

1.) Figure out why the footer panel is missing in some (many? all?) non-OSD extensions and figure out how to restore it (maybe simple, maybe not).

2.) Decide how to represent left panel content in the footer. Do we need a new icon?

3.) Decide how to refactor the current left panel code into a reusable component. Do we want the whole left panel tabbed interface in a modal window? Do we want separate icons for separate functions? Is there another option? (There are likely trade-offs here between what is technically convenient and what is the best user experience... there may be refactoring we can do to increase flexibility regardless of approach).

4.) Implement new buttons and dialogs based on decisions made in step 3, using the implementation of more info as inspiration.

**Comment by demiankatz** [comment]
Following the 2/26 standup, several of us discussed the above points and made some progress on troubleshooting and strategizing.

We found that the reason the footer is missing on most extensions is that the mobile functionality is OSD-specific. Other extensions are missing the pieces used by OSD, and so the non-mobile buttons get hidden, but there are no mobile buttons to replace them. It seems likely that this can be simplified into a universal footer panel that responds to size changes and works across all extensions.

We agreed that following the existing dialog pattern was NOT the best path forward, as it offers poor usability and actually increases technical complexity. Instead, we should work on a mechanism to fully expand panels on mobile, so that the existing panels become modal/full width and cover the content. This allows us to make the functionality available with less refactoring and less redundancy; it also allows us to simplify code by removing the existing complex logic designed to display the metadata component in a dialogue, which should make the "universal responsive footer panel" described above significantly easier to implement.

Proposed next steps:

1.) Refactor the existing "more info" dialog to use an expanded sidebar instead. This will simplify complex code and establish a pattern that can be applied to other panels.

2.) Add an "expand left panel" button that extends the functionality from step one to make the other panel available on mobile.

3.) Simplify the footer panel code into a universal responsive footer, extending the improved/consistent functionality to all extensions.

ACTION ITEMS:

@jamesmisson will begin work refactoring the more info button to expand the right panel instead of using a separate dialogue component.

@LlGC-jop will study CSS options to determine the best way to expand panels on mobile. He will coordinate with @jamesmisson so any recommendations he makes can be incorporated into the more info implementation.

@crhallberg will analyze the mobile footer panel code, to prepare for replacing it with the universal responsive footer panel.

@demiankatz will be available for brainstorming and code review as needed.

**Comment by demiankatz** [comment]
We met again today to discuss progress. @jamesmisson opened #1316 to normalize the way media queries are applied as a clean baseline for the work needed here; he will talk to @LanieOkorodudu about testing it a little more before it is merged. @LlGC-jop has begun working on a CSS refactor to make the panel layout/visibility more CSS-based instead of relying on Javascript logic. He will open a PR to make his work in progress more visible. @jamesmisson will begin working on Javascript simplification in parallel with @LlGC-jop, and the two of them will see if they can collaborate within the same PR.

**Comment by jamesmisson** [comment]
Thanks Jonathan. I think it's worth raising a PR, then @LanieOkorodudu can also do some testing too

**Comment by demiankatz** [comment]
I can likely get some testing assistance on this end later in the week as well (though my usual tester is on vacation today).

**Comment by LanieOkorodudu** [comment]
@demiankatz, I wouldn't mind to assist you here. happy to test it.

**Comment by demiankatz** [comment]
Thanks, @LanieOkorodudu, your testing is definitely appreciated! This is a big enough change that I think it's worth having more than one person look into it. On my end, we have a list of Villanova-specific manifests that cover all of our use cases, so my plan was to get help going through all of those. But of course there are other use cases that we don't use and may be less able to test. We can also compare notes and work out a strategy if that would be helpful!
